crypto/keys/pem_signer_factory.go

```diff
@@ -32,25 +33,28 @@ type PEMSignerFactory struct{}
 // pb must be one of the following types:
 // - keyspb.PEMKeyFile
 // - keyspb.PrivateKey
-func (f PEMSignerFactory) NewSigner(ctx context.Context, pb *any.Any) (crypto.Signer, error) {
-	var privateKey ptypes.DynamicAny
-	if err := ptypes.UnmarshalAny(pb, &privateKey); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal private key: %v", err)
+func (f PEMSignerFactory) NewSigner(ctx context.Context, pb proto.Message) (crypto.Signer, error) {
+	if anyPB, ok := pb.(*any.Any); ok {
+		var unmarshaledPB ptypes.DynamicAny
+		if err := ptypes.UnmarshalAny(anyPB, &unmarshaledPB); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal private key protobuf: %v", err)
+		}
+		pb = unmarshaledPB.Message
 	}
 
-	switch privateKey := privateKey.Message.(type) {
+	switch privateKey := pb.(type) {
 	case *keyspb.PEMKeyFile:
 		return NewFromPrivatePEMFile(privateKey.GetPath(), privateKey.GetPassword())
 	case *keyspb.PrivateKey:
 		return NewFromPrivateDER(privateKey.GetDer())
 	}
 
-	return nil, fmt.Errorf("unsupported PrivateKey type: %T", privateKey.Message)
+	return nil, fmt.Errorf("unsupported private key protobuf type: %T", pb)
 }
 
 // Generate creates a new private key based on a key specification.
 // It returns a proto that can be passed to NewSigner() to get a crypto.Signer.
-func (f PEMSignerFactory) Generate(ctx context.Context, spec *keyspb.Specification) (*any.Any, error) {
+func (f PEMSignerFactory) Generate(ctx context.Context, spec *keyspb.Specification) (proto.Message, error) {
```

Comment:
Not all SignerFactory implementation may want to return an actual DER-encoded private key here. A PKCS11 SignerFactory, for instance, might return a protobuf containing the key's name.